src/monodroid/jni/android-system.cc

```diff
@@ -815,17 +812,18 @@ AndroidSystem::setup_environment_from_override_file (const char *path)
 		return;
 	}
 
-	char    *buf = new char [sbuf.st_size];
-	ssize_t  nread = 0;
+	size_t   file_size = static_cast<size_t>(sbuf.st_size);
```

Comment:
As per: https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/types.h.html

`stat::st_size` is of type `off_t`, and `off_t` is a *signed* integral type.

Admittedly, it would be bona-fide *crazy* for `sbuf.st_size` to be a negative value, so I suppose we don't need to check for negative sizes...do we?  (I hope not!)

Reply:
I don't know why stat has `st_size` using a signed type (it's not like you will return an error in this field by setting it to -1 or something...), but I don't like it (as commented previously - files cannot have negative sizes) and so I opted for the more correct type for the **size**.